### PR TITLE
优化正则匹配 

### PR DESCRIPTION
**Pull request by M2shad0w**
//  m  多行模式：让 ^ 和 $ 匹配整个文本的开头和结尾，而非行首和行尾(默认为 false)

### CODE REVIEW

**Comment by siddontang** [comment]
can you add some tests?

File: canal/sync.go [hunk 1]
```diff
@@ -14,10 +14,10 @@ import (
 )
 
 var (
-	expCreateTable = regexp.MustCompile("(?i)^CREATE\\sTABLE(\\sIF\\sNOT\\sEXISTS)?\\s`{0,1}(.*?)`{0,1}\\.{0,1}`{0,1}([^`\\.]+?)`{0,1}\\s.*")
-	expAlterTable  = regexp.MustCompile("(?i)^ALTER\\sTABLE\\s.*?`{0,1}(.*?)`{0,1}\\.{0,1}`{0,1}([^`\\.]+?)`{0,1}\\s.*")
-	expRenameTable = regexp.MustCompile("(?i)^RENAME\\sTABLE\\s.*?`{0,1}(.*?)`{0,1}\\.{0,1}`{0,1}([^`\\.]+?)`{0,1}\\s{1,}TO\\s.*?")
-	expDropTable   = regexp.MustCompile("(?i)^DROP\\sTABLE(\\sIF\\sEXISTS){0,1}\\s`{0,1}(.*?)`{0,1}\\.{0,1}`{0,1}([^`\\.]+?)`{0,1}(?:$|\\s)")
+	expCreateTable = regexp.MustCompile("(?im)CREATE\\sTABLE(\\sIF\\sNOT\\sEXISTS)?\\s`{0,1}(.*?)`{0,1}\\.{0,1}`{0,1}([^`\\.]+?)`{0,1}\\s.*")
```
Comment (siddontang) [inline canal/sync.go]: can you give the comment why we need to use `(?im)` here?

Comment (M2shad0w) [reply canal/sync.go]:
```
INFO[0218] row data [47 42 32 65 112 112 108 105 99 97 116 105 111 110 78 97 109 101 61 80 121 67 104 97 114 109 32 50 48 49 55 46 50 46 52 32 42 47 32 65 76 84 69 82 32 84 65 66 76 69 32 99 111 109 109 117 110 105 116 121 95 99 111 109 98 105 110 101 32 67 72 65 78 71 69 32 96 50 50 50 49 51 50 96 32 102 115 100 118 102 32 73 78 84 40 49 49 41]
 query data /* ApplicationName=PyCharm 2017.2.4 */ ALTER TABLE community_combine CHANGE `222132` fsdvf INT(11) 
```

Comment (M2shad0w) [reply canal/sync.go]: query event bin data is like `/* ApplicationName=PyCharm 2017.2.4 */ ALTER TABLE community_combine CHANGE `222132` fsdvf INT(11) `
if not use reg 多行模式，只匹配行首就不能正确解析事件了

Comment (siddontang) [reply canal/sync.go]: can you add a test for this?

**Comment by GregoryIan** [comment]
#382 is the better way, I think we should close this pr, still thanks @M2shad0w

**Pull request closed without merging**